### ISSUE TITLE
[question] conan install behavior with path vs. reference

### ISSUE THREAD

**Issue by ineb** [conan-io/conan#10148]
My goal is to achieve the behavior of `conan create` but with the created packaged being created at a defined location instead of the local cache.

I looked at the `conan install` command but see different behavior when using paths or references.

What works is:
```
conan create . user/channel -o my_package:some_setting=True --build outdated
conan install my_package/0.0.1@user/channel -o my_package:some_setting=True --install-folder=pkg_folder
```
Inside `pkg_folder/` is see the expected binaries of `my_package`.

But when using:
```
conan install . -o my_package:some_setting=True --install-folder=pkg_folder --build outdated
```
Inside `pkg_folder/` i only see files like `conanbuildinfo.cmake`, `conaninfo.txt` but nothing has been built.

The [documentation](https://docs.conan.io/en/latest/reference/commands/consumer/install.html) of `conan install` says:
```
conan install executes methods of a conanfile.py in the following order:

    config_options()
    configure()
    requirements()
    package_id()
    package_info()
    deploy()

Note this describes the process of installing a pre-built binary package. If the package has to be built, conan install --build executes the following:

    config_options()
    configure()
    requirements()
    package_id()
    build_requirements()
    build_id()
    system_requirements()
    source()
    imports()
    build()
    package()
    package_info()
    deploy()
```
I thought using `--build` would then call the method chain as documented but it looks like it does not when a path is used as argument.

I need this because with `conan create` the `deploy()` method is called to copy package dependencies alongside the binaries. This way, the final product can be shipped.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
The only way to build a package in the user folder is when you have the recipe of that package locally. Then doing ``conan install`` + ``conan build`` (or local ``cmake ...`` or other build system) would build locally that package. If the ``conan create`` works, then ``conan install . `` + ``conan build . `` should work too and build locally (given the layout is correct, etc). If you can elaborate with more details why this would fail in your case, that would be useful.

**Comment by ineb** [comment]
`conan install .` + `conan build .` does work in my case but i am missing the call to `deploy()` in this scenario.
What i need would be `conan install .` + `conan build .` + `conan deploy .`

I understand that conan is not mainly about deployment. Having a `conan deploy` command which calls the `deploy()` method from the (local) `conanfile.py` would be nice though.

**Comment by dornbirndevelops** [comment]
I see two behaviors for the install command, though they have the same naming (can be confusing at first).
- `conan install path/to/conanfile.py ...` installs the dependencies required by the local recipe. this is useful when you want to set up a package development environment.
- `conan install package/version@user/channel ...` installs the binaries of this package into the cache. by also specifying the installation folder here, the specified folder becomes the target of the actions specified in deploy.

In a nutshell the two usages (according to my personal workflow):
1. work on a package --> `conan install <recipe-path>` + `conan build <recipe-path>` + (`conan export-pkg <recipe-path>`)
2. install binaries of your package (maybe to targets that don't have conan) --> `conan install <package-reference> -if <path-i-want>`

**Comment by memsharded** [comment]
They are indeed 2 different cases.

- ``conan install <path>`` uses the conanfile at that path as the "root" of the graph. That install the dependencies of that conanfile, not the conanfile itself (which doesn't even require a name or a version)
- ``conan install <reference>`` uses a "virtual" consumer nood with a ``requires=<reference>``, to force depending on it. It is the equivalent of writing a conanfile.txt with such a single ``requires`` and doing ``conan install .`` over it. This is necessary because when you do that, you want to consume that reference, as a requires.

**Comment by memsharded** [comment]
Closing as responded back then.
Also this ticket is quite outdated, closing as outdated too. Everything has changed since then, including the command line commands and arguments.

Please report new tickets against Conan 2 if necessary, thanks for your feedback.
